### PR TITLE
Make custom emails

### PR DESCRIPTION
**Pull request by zhoua0805**
**NOTE**: This PR comes after PR #63 

- Customize email recipient and subject line
- Attach itinerary content from html to txt file
- Hide the send-email button if itinerary content is empty

### CODE REVIEW

File: planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java [hunk 1]
```diff
@@ -34,47 +35,60 @@
...
-        try {
-            Message message = new MimeMessage(session);
-            message.setFrom(new InternetAddress("alicexyz@google.com", "Alice Zhou"));
-            //TODO: change this to the signed in user's email address
-            message.addRecipient(Message.RecipientType.TO,
-                            new InternetAddress("alicexyz@google.com"));
-            //TODO: add custom subject, html body, and attached file
-            message.setSubject("Your itinerary for Toronto trip");
-            Multipart multipartContent = new MimeMultipart();
+        BufferedReader br = request.getReader();
```
Comment (azhang72) [inline planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java]: nit: avoid using abbreviations.

File: planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java [hunk 2]
```diff
@@ -34,47 +35,60 @@
...
-            htmlPart.setContent(htmlBody, "text/html");
-            multipartContent.addBodyPart(htmlPart);
+        if (br != null) {
+            itineraryContent =  br.lines().collect(Collectors.joining("\n"));
+            Properties props = new Properties();
+            Session session = Session.getDefaultInstance(props, null);
+            
+            try {
+                Message message = new MimeMessage(session);
+                message.setFrom(new InternetAddress("alicexyz@google.com", "Alice Zhou"));
```
Comment (azhang72) [inline planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java:58]: nit: add a TODO to set up an email for Planet? :P

File: planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java [hunk 3]
```diff
@@ -34,47 +35,60 @@
...
+            } catch (MessagingException e) {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                response.getWriter().println(e.getMessage());
+            } catch (UnsupportedEncodingException e) {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                response.getWriter().println(e.getMessage());
+            } 
+        } else {
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            response.getWriter().println("");
```
Comment (azhang72) [inline planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java]: Is this necessary?

File: planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java (hunk 1 above)
Comment (zhoua0805) [reply planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java]: Okay

File: planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java (hunk 2 above)
Comment (zhoua0805) [reply planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java:58]: Okay!

File: planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java (hunk 3 above)
Comment (zhoua0805) [reply planet/src/main/java/com/google/planet/servlets/SendEmailServlet.java]: Not really... I just removed it, thank you!

**Review by nabil-dbz — APPROVED** [review]

**Review by azhang72 — APPROVED** [review]

**Pull request merged**